ui-v2/app/templates/dc/intentions/index.hbs

```diff
@@ -1,79 +1,83 @@
+    {{#let (selectable-key-values
+        (array "" (concat "All (" items.length ")"))
+        (array "allow" (concat "Allow (" (sub items.length denied.length) ")"))
+        (array "deny" (concat "Deny (" denied.length ")"))
+          selected=filterBy
+      )
+      as |filter|
+    }}
+    <AppView @class="intention list" @loading={{not loaded}}>
+      <BlockSlot @name="header">
+          <h1>
+              Intentions <em>{{format-number items.length}} total</em>
+          </h1>
+          <label for="toolbar-toggle"></label>
+      </BlockSlot>
+      <BlockSlot @name="actions">
+          <a data-test-create href="{{href-to 'dc.intentions.create'}}" class="type-create">Create</a>
+      </BlockSlot>
+      <BlockSlot @name="toolbar">
+  {{#if (gt items.length 0) }}
+        <SearchBar
+          data-test-intention-filter="true"
+          @value={{search}}
+          @onsearch={{action (mut search) value="target.value"}}
+          @selected={{filter.selected}}
+          @options={{filter.items}}
+          @onchange={{action (mut filterBy) value='target.value'}}
+        />
+  {{/if}}
+      </BlockSlot>
+      <BlockSlot @name="content">
+          <ChangeableSet @dispatcher={{searchable 'intention' (if (eq filter.selected.key "") items (filter-by "Action" filter.selected.key items))}} @terms={{search}}>
+            <BlockSlot @name="content" as |filtered|>
+              <ConsulIntentionList
+                @items={{filtered}}
+                >
+                  <EmptyState @allowLogin={{true}}>
+                    <BlockSlot @name="header">
+                      <h2>
+                        {{#if (gt items.length 0)}}
+                          No intentions found
+                        {{else}}
+                          Welcome to Intentions
+                        {{/if}}
+                      </h2>
+                    </BlockSlot>
+                    <BlockSlot @name="body">
+                      <p>
+                        {{#if (gt items.length 0)}}
+                          No intentions where found matching that search, or you may not have access to view the intentions you are searching for.
+                        {{else}}
+                          There don't seem to be any intentions, or you may not have access to view intentions yet.
+                        {{/if}}
+                      </p>
+                    </BlockSlot>
+                    <BlockSlot @name="actions">
+                      <li class="docs-link">
+                        <a href="{{env 'CONSUL_DOCS_URL'}}/commands/intention" rel="noopener noreferrer" target="_blank">Documentation on intentions</a>
+                      </li>
+                      <li class="learn-link">
+                        <a href="{{env 'CONSUL_DOCS_LEARN_URL'}}/consul/getting-started/connect" rel="noopener noreferrer" target="_blank">Read the guide</a>
+                      </li>
+                    </BlockSlot>
+                  </EmptyState>
```

Comment:
Looking at this I'm wondering if we should use block-slots here, and have say a `body` and `empty` slot? That way it makes it more explicit what this is doing 🤔 . I could probably switch this out in a later PR.

Reply:
🤔 block-slots are already used here...

Reply:
Oh sorry, yeah this comment is a little off/misplaced. Should have tried to get it closer to the `<ConsulIntentionList />`. Just to clear up, its currently like this:

```hbs
<ConsulIntentionList>
  <p>The empty state</p>
</ConsulIntentionList>
```

I was thinking of making it this

```hbs
<ConsulIntentionList>
  <BlockSlot @name="empty">
   <p>The empty state</p>
  </BlockSlot>
</ConsulIntentionList>
```

...which I think just makes it clearer what the content of the ConsulIntentionList component is used for.

We could choose a default empty state for the full set of 'listing' components also